### PR TITLE
Replaced unsafe string function

### PR DESCRIPTION
**Pull request by shrikant1407**
* Replaced unsafe string function
* Enable debug logs for fdo_sys module

Signed-off-by: Shrikant Temburwar <shrikant.temburwar@intel.com>

### CODE REVIEW

File: device_modules/fdo_sys/fdo_sys.c [hunk 1]
```diff
@@ -75,7 +75,7 @@ int fdo_sys(fdo_sdk_si_type type,
 			fdow = ModuleAlloc(sizeof(fdow_t));
 			if (!fdow_init(fdow) || !fdo_block_alloc_with_size(&fdow->b, MOD_MAX_BUFF_SIZE)) {
 #ifdef DEBUG_LOGS
```
Comment (tajnisha) [inline device_modules/fdo_sys/fdo_sys.c:77]: Any reason why are we still retaining teh 'ifdef DEBUG_LOGS' still ?

File: device_modules/fdo_sys/fdo_sys.c [hunk 2]
```diff
@@ -75,7 +75,7 @@ int fdo_sys(fdo_sdk_si_type type,
...
-				printf(
+				LOG(LOG_DEBUG,
 					"Module fdo_sys - FDOW Initialization/Allocation failed!\n");
 #endif
```
Comment (tajnisha) [inline device_modules/fdo_sys/fdo_sys.c:80]: Any reason why are we still retaining teh '#endif' still ? I guess we can remove it.

File: storage/util.c [hunk 3]
```diff
@@ -234,6 +234,40 @@ void *fdo_alloc(size_t size)
...
+/**
+ * Internal API to create string duplicate
+ */
+char *strdup_s(const char *str) {
+
+	if (NULL == str) {
+		return NULL;
+	}
+
+	int len = strnlen_s(str, FDO_MAX_STR_SIZE) + 1;
```
Comment (tajnisha) [inline storage/util.c:247]: Can this be updated as unsigned int ?

File: storage/util.c [hunk 4]
```diff
@@ -234,6 +234,40 @@ void *fdo_alloc(size_t size)
...
+
+	if (memset_s(buf, len, 0) != 0) {
+		LOG(LOG_ERROR, "Memset Failed\n");
+		fdo_free(buf);
+		goto end;
+	}
+
+	if (memcpy_s(buf, len - 1, str, len - 1)) {
+		LOG(LOG_ERROR, "Copying cleartext failed\n");
+		goto end;
```
Comment (tajnisha) [inline storage/util.c:264]: add fdo_free(buf) if copy failed ?

**Review by tajnisha — APPROVED** [review]

**Review by v-vasavi — APPROVED** [review]

**Pull request merged**